d2compiler/compile.go

```diff
@@ -277,6 +277,16 @@ func (c *compiler) compileField(obj *d2graph.Object, f *d2ir.Field) {
 			}
 		}
 		return
+	} else if f.Name == "vars" {
+		if f.Map() != nil {
+			if len(f.Map().Edges) > 0 {
+				c.errorf(f.Map().Edges[0].LastRef().AST(), "vars cannot contain an edge")
+			}
+		}
+		if d2ir.NodeBoardKind(d2ir.ParentMap(f)) == "" {
+			c.errorf(f.LastRef().AST(), "vars must be defined at the root of a board")
```

Comment:
Why is this a requirement? I wanted vars to be able to be defined on a per map basis rather than per board.